bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/parallacticangle/ParallacticAngleControls.scala

```diff
@@ -142,32 +143,42 @@ class ParallacticAngleControls(isPaUi: Boolean) extends GridBagPanel with Publis
         * This should be called whenever the position angle or parallactic angle changes.
         * A warning icon and tooltip are displayed if the two are different.
         */
-      def warningState(): Unit = Swing.onEDT {
+      def warningState(): Unit =
         for {
           e     <- editor
           fmt   <- formatter
           inst  <- Option(e.getContextInstrumentDataObject).collect { case s: SPInstObsComp => s }
         } warningStateFromPAString(inst.getPosAngleDegrees.toString)
-      }
 
-      def warningStateFromPAString(paStr: String): Unit = Swing.onEDT {
+      def warningStateFromPAString(paStr: String): Unit =
         for {
           e     <- editor
           fmt   <- formatter
           inst  <- Option(e.getContextInstrumentDataObject).collect { case s: SPInstObsComp => s }
-          angle <- parallacticAngle
         } {
-          val explicitlySet = !fmt.format(angle.toDegrees).equals(paStr) &&
-                              !fmt.format((angle + Angle.fromDegrees(180)).toDegrees).equals(paStr)
-          if (explicitlySet) {
+          // We warn only if the parallactic angle exists and has been explicitly set.
+          val warningFlag = parallacticAngle.exists { angle =>
+              val fa = fmt.format(angle.toDegrees)
+              val faFlip = fmt.format(angle.flip.toDegrees)
+              !fa.equals(paStr) && !faFlip.equals(paStr)
+          }
+
+          if (warningFlag) {
             icon = Resources.getIcon("eclipse/alert.gif")
```

Comment:
😸  After last week, I think I am still having deep fears about the `Swing.onEDT`. I'm pretty sure these are all invoked through methods scheduled on the EDT, but I will try only doing them on the EDT and see what happens. I'm a bit hesitant about not immediately doing the operation, since it can involve the contents of the PA text field, and again, we have the issue of this updating from multiple sources.

I'll play around and dig around a bit and see.